index.js

```diff
@@ -117,9 +124,11 @@ var salita = function salita(dir, options, callback) {
     var depLookups = [];
     var depPromises = [];
     forEach(deps, function (title, key) {
-      var depLookup = Promise.all(dependenciesLookup(pkg.data, key, options['ignore-stars'], options['ignore-pegged']));
-      depLookups.push(depLookup);
-      depPromises.push(depLookup.then(options.json ? createResultJSON(key) : createResultTable(title)));
+      if (sectionCheck(options, key)) {
```

Comment:
this can probably just pass `options.section`; there's no need to pass the entire object here.